packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -44,6 +45,8 @@ const knownIgnoreList = new Set(['/', '/favicon.ico'])
 export function transformMiddleware(
   server: ViteDevServer,
 ): Connect.NextHandleFunction {
+  // cache moduleGraph, as server.moduleGraph references a clean graph that this middleware shouldn't
+  // read or modify after a restart
```

Comment:
This comment seems to be unneeded now? 🤔

Reply:
Removed. Also reworded another comment here: [`19150d8` (#13262)](https://github.com/vitejs/vite/pull/13262/commits/19150d8e07f46f5d19789554144dd7b4b8fa9b94). We could actually revert https://github.com/vitejs/vite/pull/13231, but I feel it is better to leave the extended condition for now.